.github/workflows/release_nightly.yml

```diff
@@ -486,46 +452,102 @@ jobs:
-          path: demo
-          ref: master
-          fetch-depth: 0
-          persist-credentials: false # Needed to allow commit via RUFFLE_BUILD_TOKEN below
+          toolchain: stable
+          targets: wasm32-unknown-unknown
+          components: rust-src
 
-      - name: Update web demo
-        if: matrix.demo
-        run: |
-          # Delete the old build.
-          rm -fr *
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22"
+          registry-url: https://registry.npmjs.org
 
-          # Copy the fresh build into this folder.
-          cp -fr ../web/packages/demo/dist/* .
+      # wasm-bindgen-cli version must match wasm-bindgen crate version.
+      # Be sure to update in test_web.yml, Cargo.toml, web/docker/Dockerfile,
+      # and web/README.md as well.
+      - name: Install wasm-bindgen
+        run: cargo install wasm-bindgen-cli --version 0.2.100
 
-          # Restore our custom swfs
-          git restore swfs swfs.json
+      # Keep the version number in sync in all workflows,
+      # and in the extension builder Dockerfile!
+      - name: Install wasm-opt
+        uses: sigoden/install-binary@v1
+        with:
+          repo: WebAssembly/binaryen
+          tag: version_122
+          name: wasm-opt
 
-          # Create git commit. Amend previous commit to avoid daily commit spam.
-          git config user.name "RuffleBuild"
-          git config user.email "ruffle@ruffle.rs"
-          git add -A
-          git commit --amend -m "Nightly build ${{ needs.create-nightly-release.outputs.date }}"
-        working-directory: demo
+      - name: Install node packages
+        working-directory: web
+        shell: bash -l {0}
+        run: npm ci
 
-      - name: Push web demo
-        if: github.repository == 'ruffle-rs/ruffle' && matrix.demo
-        uses: ad-m/github-push-action@master
-        with:
-          repository: ruffle-rs/demo
-          github_token: ${{ secrets.RUFFLE_BUILD_TOKEN }}
-          directory: demo
-          force: true
+      - name: Seal version data
+        shell: bash -l {0}
+        working-directory: web
+        env:
+          BUILD_ID: ${{ github.run_number }}
+          ENABLE_VERSION_SEAL: "true"
+          FIREFOX_EXTENSION_ID: ${{ secrets.FIREFOX_EXTENSION_ID }} # Needed to inject into manifest.json
+        run: npm run version-seal
+
+      - name: Build web
+        env:
+          BUILD_ID: ${{ github.run_number }}
+          # Build web demo with WebGPU support for testing in Chrome origin trial on ruffle.rs
+          CARGO_FEATURES: jpegxr,webgpu
+          FIREFOX_EXTENSION_ID: ${{ secrets.FIREFOX_EXTENSION_ID }} # Needed to inject into manifest.json
+          WASM_SOURCE: cargo_and_store
+        working-directory: web
+        shell: bash -l {0}
+        run: npm run build:repro
+
+      - name: Build web docs
+        working-directory: web
+        run: npm run docs
```

Comment:
This isn't a change from the current workflows even, but while we're looking at this, do we know why both workflows build the web docs?

Reply:
Seems unnecessary unless I misunderstand something.